src/components/TeacherComponents/UnitsLearningThemeFilters/UnitsLearningThemeFilters.tsx

```diff
@@ -25,56 +26,77 @@ export type UnitsLearningThemeFiltersProps = {
   learningThemes: LearningTheme[] | null;
   linkProps: UnitListingLinkProps | SpecialistUnitListingLinkProps;
   trackingProps?: LearningThemeSelectedTrackingProps;
+  idSuffix: string;
 };
+
 const UnitsLearningThemeFilters = ({
-  labelledBy,
   learningThemes = [],
   selectedThemeSlug,
   linkProps,
   trackingProps,
+  idSuffix,
 }: UnitsLearningThemeFiltersProps) => {
   const [skipFiltersButton, setSkipFiltersButton] = useState(false);
-  const listStateProps = useCategoryFilterList({
-    selectedKey: selectedThemeSlug,
-    getKey: (
-      linkProps: UnitListingLinkProps | SpecialistUnitListingLinkProps,
-    ) => {
-      if (linkProps.search?.["learning-theme"]) {
-        return linkProps.search?.["learning-theme"];
-      } else return "all";
-    },
-  });
-
-  const learningThemesMapped = learningThemes
+  const learningThemesMapped: Array<RadioTheme> = learningThemes
     ? learningThemes
         .map((learningTheme) => {
           return {
             label: learningTheme?.themeTitle,
             slug: learningTheme?.themeSlug,
           };
         })
-        .sort(
-          (
-            a: {
-              label: string | undefined | null;
-              slug: string | undefined | null;
-            },
-            b: {
-              label: string | undefined | null;
-              slug: string | undefined | null;
-            },
-          ) => {
-            if (a?.slug === "no-theme") {
-              return 0;
-            } else if (b?.slug === "no-theme") {
-              return -1;
-            } else {
-              return 0;
-            }
-          },
-        )
+        .filter((theme) => isRadioTheme(theme))
+        .map((theme) => theme as RadioTheme)
+        .sort((a: RadioTheme, b: RadioTheme) => {
```

Comment:
```suggestion
        .filter(isRadioTheme)
        .sort((a, b) => {
```

the extra `=>`, `map` and `sort` typing should be unnecessary as type info is supposed to be forwarded through `isRadioTheme`.

Reply:
you would think! this drives me mad, but it won't infer the type without an assertion here - actually though I think we can add that as the filter return type, which would be a bit neater: `.filter((theme): theme is RadioTheme => isRadioTheme(theme))`

Reply:
Weird. Works on my machine 🤣 

<img width="448" alt="Screenshot 2024-09-17 at 11 42 49" src="https://github.com/user-attachments/assets/bc2cea49-df45-4a0c-8aca-f682e820b9e1">

Reply:
now hang on, I've been tricked!
![Screenshot 2024-09-17 at 11 45 23](https://github.com/user-attachments/assets/e1a7207d-6ebf-4823-81f7-120fdaea76f9)
![Screenshot 2024-09-17 at 11 45 54](https://github.com/user-attachments/assets/f331e3c7-4c81-4298-94d3-327bf28e30ad)